File: VotingApp_Client_WEB/Models/VotingSessionEgress.js
// Define the VotingSessionEgress class
export class VotingSessionEgress {
    constructor() {
      this._sessionTitle = "";
      this._creator = "";
      this._questions = [];
    }
  
    get SessionTitle() {
      return this._sessionTitle;
    }
  
    set SessionTitle(value) {
      this._sessionTitle = value;
    }
  
    get Creator() {
      return this._creator;
    }
  
    set Creator(value) {
      this._creator = value;
    }
  
    get Questions() {
      return this._questions;
    }
  
    set Questions(value) {
      this._questions = value;
    }

        // Method to convert the object to JSON with custom property names
        toJSON() {
            return {
                title: this._sessionTitle,
                creator: this._creator,
                questions: this._questions
            };
        }
  }
  

File: VotingApp_Client_WPF/VotingSessionEgress.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace VotingApp_Client_WPF
{
    // container class for session data outgoing
    public class VotingSessionEgress
    {
        private string? _sessionTitle;
        private string? _creator;
        private List<VotingQuestion?>? _questions = new();

        public VotingSessionEgress() { }

        [JsonPropertyName("title")]
        public string? SessionTitle { get { return _sessionTitle; } set { _sessionTitle = value; } }
        [JsonPropertyName("creator")]
        public string? Creator { get { return _creator; } set { _creator = value; } }
        [JsonPropertyName("questions")]
        public List<VotingQuestion?>? Questions { get { return _questions; } set { _questions = value; } }
    }
}
